services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -933,7 +933,7 @@ public void joinGroup(@PathParam("groupId") String groupId) {
             throw new NotFoundException("Group not found");
         }
         auth.groups().requireManageMembership(group);
-        if (!user.isMemberOf(group)){
+        if (user.getGroupsStream().noneMatch(g -> group.getId().equals(g.getId()))){
```

Comment:
Could you please for the sake of readability leverage the newly introduced `RoleUtils.isDirectMember` method here as well?

Reply:
Done.